src/main/java/spark/ExceptionHandlerImpl.java

```diff
@@ -16,18 +16,18 @@
  */
 package spark;
 
-public abstract class ExceptionHandlerImpl {
+public abstract class ExceptionHandlerImpl<T extends Exception> {
     /**
      * Holds the type of exception that this filter will handle
      */
-    protected Class<? extends Exception> exceptionClass;
+    protected Class<? extends T> exceptionClass;
```

Comment:
Changed it back, because otherwise the exceptionClass get and set methods do not work as before using the default case (t = Exception)